airbyte-webapp/src/views/Connector/ConnectorForm/useBuildForm.tsx

```diff
@@ -41,16 +41,51 @@ export function useBuildForm(
     }),
     [formType, formatMessage, selectedConnectorDefinitionSpecification.connectionSpecification]
   );
-  const startValues = useMemo<ConnectorFormValues>(
-    () => ({
+
+  const formFields = useMemo<FormBlock>(() => jsonSchemaToFormBlock(jsonSchema), [jsonSchema]);
+
+  if (formFields._type !== "formGroup") {
+    throw new FormBuildError("Top level configuration has to be an object");
+  }
+
+  const startValues = useMemo<ConnectorFormValues>(() => {
+    if (isEditMode) {
+      return {
+        name: "",
+        connectionConfiguration: {},
+        ...initialValues,
+      };
+    }
+    const baseValues = {
       name: "",
       connectionConfiguration: {},
       ...initialValues,
-    }),
-    [initialValues]
-  );
+    };
+
+    function setDefaultValues(formGroup: FormGroupItem, values: Record<string, unknown>) {
```

Comment:
As it's only called in here recursively this seemed cleaner to me (keeps it closer to the place where it's used). No strong opinion on these things though, let me know if you would prefer to move it.

Reply:
I'm partial to keeping unwrapped (non-useCallback, etc) methods at the top scope within the react render tree since every time this is called (which may be a small number of times) it's redefined.

Reply:
Wouldn't matter in this specific case, but I agree it's a good default. Improved